uptrain/operators/language/sql.py

```diff
@@ -0,0 +1,185 @@
+            except ParseError:
+                tables.append(json.dumps({}))
+                is_valid.append(False)
+
+        return {"output": data.with_columns([pl.Series(self.col_out_tables, tables),
+                                             pl.Series(self.col_out_is_valid_sql, is_valid)])}
+
+
+# Ensures that table names from response are valid tables
+@register_op
+class ValidateTables(TableOp):
+    col_in_response_tables: str = "response_tables"
+    col_in_schema_tables: str = "schema_tables"
+    col_out_tables_valid: str = "tables_valid"
+    col_out_cols_valid: str = "cols_valid"
+
+    def setup(self, _: t.Optional[Settings] = None):
+        return self
+
+    def run(self, data: pl.DataFrame) -> TYPE_TABLE_OUTPUT:
+        response_tables = data.get_column(self.col_in_response_tables)
+        schema_tables = data.get_column(self.col_in_schema_tables)
+        results = []
+        results_column = []
+        for response_table, schema_table in zip(response_tables, schema_tables):
+            is_valid_table = True
+            is_valid_column = True
+            # deserialize json
+            s = json.loads(schema_table)
+            r = json.loads(response_table)
+            columns_list = [columns for table, columns in s.items()]
+            all_columns = set(itertools.chain(*columns_list))
+            for table, columns in r.items():
+                is_valid_table &= table == PLACEHOLDER_TABLE or table in s
+                # TODO: handle aliases and do lineage check
+                is_valid_column = is_valid_column and is_valid_table and \
+                                   ((table != PLACEHOLDER_TABLE and set(columns).issubset(set(s[table]))) or
+                                    (table == PLACEHOLDER_TABLE and set(columns).issubset(all_columns)))
+
+            results.append(is_valid_table)
+            results_column.append(is_valid_column)
+        return {"output": data.with_columns(
+            [pl.Series(self.col_out_tables_valid, results), pl.Series(self.col_out_cols_valid, results_column)])}
+
+
+@register_op
+class ExecuteSQL(TableOp):
+    col_in_response_sql: str = "response_sql"
+    col_in_gt_sql: str = "gt_query"
+    col_in_db_path: str = "db_path"
+    col_out_execution_accuracy: str = "execution_accuracy"
+
+    def setup(self, settings: t.Optional[Settings] = None):
+        return self
+
+    def run(self, data: pl.DataFrame) -> TYPE_TABLE_OUTPUT:
+        response_sqls = data.get_column(self.col_in_response_sql)
+        gt_sqls = data.get_column(self.col_in_gt_sql)
+        db_paths = data.get_column(self.col_in_db_path)
+        results = []
+        for response_sql, gt_sql, db_path in zip(response_sqls, gt_sqls, db_paths):
+            results.append(execute_and_compare_sql(response_sql, gt_sql, db_path))
+        return {"output": data.with_columns([pl.Series(self.col_out_execution_accuracy, results)])}
+
+
+# Check if SQL has star
+@register_op
+class HasStar(TableOp):
```

Comment:
Generalise to KeywordDetector? where * can be part of the config and put it in operators/language/text.py